src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs

```diff
@@ -748,5 +749,38 @@ public async Task<T> TestWithAsyncGenericReturnType<T>(T arg1)
         }
 
         #endregion
+
+        [TestCase(TypeArgs = new[] { typeof(int) })]
+        public void ExplicitTypeArgsWithoutParameters<T>()
+        {
+            Assert.That(typeof(T), Is.EqualTo(typeof(int)));
+        }
+
+        [TestCase(2, TypeArgs = new[] { typeof(int) })]
+        public void ExplicitTypeArgsWithUnrelatedParameters<T>(int input)
```

Comment:
I think the name of the method is wrong as `2` is `typeof(int)`
This seems more the test that belongs to the `ExplicitTypeArgsWithCompatibleParameters` method below.

Can you add an practical example where the default determined type argument is wrong?
The only one I can think is a class which has an (implicit) conversion from string, e..g. DateTime.

Reply:
Very true. My thinking behind the name was that the generic method type and the argument type are "unrelated" from the compiler's perspective - as opposed to if `input` were type `T` as well. Having them still be the same underlying type leaves something to be desired though.

My thinking behind this test and the one above where `input` _is_ of type `T` is that they represent a change in responsibilities internally. Previously, or presently but without passing `TypeArgs`, it is up to the framework to infer what generic types to use for the method and it does this through inference based on the values passed into the test case as arguments. Now, when `TypeArgs` is specified, the responsibility for that shifts to the user as we will always use their specified types for the generic method - even if it conflicts with the values they try to explicitly pass in! This last scenario is the `MethodWithIncompatibleTypeArgs<T>(T input)` test added.

I'll try to revisit some of these - in particular the types we pass - to make it a bit clearer and robust.